app/scripts/modules/kubernetes/src/v2/pipelines/stages/ManifestCopier.tsx

```diff
@@ -0,0 +1,222 @@
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { Option } from 'react-select';
+import { IPromise } from 'angular';
+import { groupBy, sortBy } from 'lodash';
+
+import {
+  ModalClose,
+  TetheredSelect,
+  AccountTag,
+  Application,
+  IServerGroup,
+  noop,
+  IManifest,
+  IServerGroupManager,
+  robotToHuman,
+  ManifestReader,
+  ISecurityGroup,
+  ILoadBalancer,
+} from '@spinnaker/core';
+
+export interface IManifestCopierProps {
+  application: Application;
+  cloudProvider: string;
+  show: boolean;
+  onDismiss: () => void;
+  onManifestSelected: (manifest: IManifest) => void;
+}
+
+export interface IManifestCopierState {
+  selectedManifest: IManifestOption;
+  manifests: IManifestOption[];
+}
+
+export interface IManifestOption {
+  account: string;
+  location: string;
+  name: string;
+  kind: string;
+  data: any;
+  key: string;
+}
+
+const LAST_APPLIED_CONFIGURATION = 'kubectl.kubernetes.io/last-applied-configuration';
+
+/*
+ * A modal that allows a user to copy a running Kubernetes resource.
+ **/
+export class ManifestCopier extends React.Component<IManifestCopierProps, IManifestCopierState> {
+  public static getDerivedStateFromProps = (props: IManifestCopierProps): IManifestCopierState => {
+    let manifests: IManifestOption[] = [];
+
+    const serverGroups: IServerGroup[] = props.application.getDataSource('serverGroups').data.filter(
```

Comment:
Why only server groups?

Reply:
What do you mean? They're just handled a little differently, but the other kinds are included...

Reply:
Sorry got sidetracked halfway through. I second (from your todo) using the account map to derive kinds to load